// @wordpress/components/build/ui/card/styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.xSmall = exports.medium = exports.small = exports.rounded = exports.borderless = exports.borderRadius = exports.headerFooter = exports.InnerBody = exports.Body = exports.Content = exports.Header = exports.Card = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styles = require("@wp-g2/styles");

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tmin-height: 24px;\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2.default)([""]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tmin-height: 30px;\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tborder-radius: ", ";\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbox-shadow: none;\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\t&:first-of-type {\n\t\tborder-top-left-radius: ", ";\n\t\tborder-top-right-radius: ", ";\n\t}\n\n\t&:last-of-type {\n\t\tborder-bottom-left-radius: ", ";\n\t\tborder-bottom-right-radius: ", ";\n\t}\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tborder-color: ", ";\n\tmin-height: ", ";\n\tpadding-bottom: ", ";\n\tpadding-left: ", ";\n\tpadding-right: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tmargin-left: calc( ", " * -1 );\n\tmargin-right: calc( ", " * -1 );\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\theight: auto;\n\tmax-height: 100%;\n\tpadding: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\theight: 100%;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tborder-bottom: 1px solid;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbox-shadow: 0 0 0 1px ", ";\n\toutline: none;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Card = (0, _styles.css)(_templateObject(), _styles.ui.get('surfaceBorderColor'));
exports.Card = Card;
var Header = (0, _styles.css)(_templateObject2());
exports.Header = Header;
var Content = (0, _styles.css)(_templateObject3());
exports.Content = Content;
var Body = (0, _styles.css)(_templateObject4(), _styles.ui.get('cardPadding'));
exports.Body = Body;
var InnerBody = (0, _styles.css)(_templateObject5(), _styles.ui.get('cardPadding'), _styles.ui.get('cardPadding'));
exports.InnerBody = InnerBody;
var headerFooter = (0, _styles.css)(_templateObject6(), _styles.ui.get('colorDivider'), _styles.ui.get('cardHeaderHeight'), _styles.ui.get('cardHeaderFooterPaddingY'), _styles.ui.get('cardPaddingX'), _styles.ui.get('cardPaddingX'), _styles.ui.get('cardHeaderFooterPaddingY'));
exports.headerFooter = headerFooter;
var borderRadius = (0, _styles.css)(_templateObject7(), _styles.ui.get('cardBorderRadius'), _styles.ui.get('cardBorderRadius'), _styles.ui.get('cardBorderRadius'), _styles.ui.get('cardBorderRadius'));
exports.borderRadius = borderRadius;
var borderless = (0, _styles.css)(_templateObject8());
exports.borderless = borderless;
var rounded = (0, _styles.css)(_templateObject9(), _styles.ui.get('cardBorderRadius'));
exports.rounded = rounded;
var small = (0, _styles.css)(_templateObject10());
exports.small = small;
var medium = (0, _styles.css)(_templateObject11());
exports.medium = medium;
var xSmall = (0, _styles.css)(_templateObject12());
exports.xSmall = xSmall;
