// @wordpress/components/build-module/ui/control-label/styles.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n\tdisplay: block;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n\tdisplay: inline-block;\n\tvertical-align: middle;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n\tdisplay: inline-block;\n\tline-height: ", ";\n\tmargin: 0;\n\tmax-width: 100%;\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n\n\t&:active {\n\t\tuser-select: none;\n\t}\n\n\t", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n\t> * {\n\t\tposition: relative;\n\t\ttop: 0.5px;\n\t}\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

/**
 * External dependencies
 */
import { css, getHighDpi, ui } from '@wp-g2/styles';
var lineHeight = "calc(".concat(ui.get('fontSize'), " * 1.2)");
/* eslint-disable jsdoc/valid-types */

/**
 * @param {Parameters<typeof ui.get>[0]} size The padding size.
 */

/* eslint-enable jsdoc/valid-types */

function getPadding(size) {
  return "calc((".concat(ui.get(size), " - ").concat(lineHeight, ") / 2)");
}

var highDpiAdjust = getHighDpi(_templateObject());
export var ControlLabel = css(_templateObject2(), lineHeight, getPadding('controlHeight'), getPadding('controlHeight'), highDpiAdjust);
export var large = css(_templateObject3(), getPadding('controlHeightLarge'), getPadding('controlHeightLarge'));
export var medium = css(_templateObject4(), getPadding('controlHeight'), getPadding('controlHeight'));
export var small = css(_templateObject5(), getPadding('controlHeightSmall'), getPadding('controlHeightSmall'));
export var inline = css(_templateObject6());
export var block = css(_templateObject7());
