// @wordpress/components/build-module/ui/button/styles.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";

function _templateObject19() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n\t", ";\n"]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteral(["\n\t&[data-active='true'] {\n\t\t", ";\n\t\t", ";\n\n\t\tbackground: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\n\t\t&:hover {\n\t\t\tbackground: ", ";\n\t\t}\n\t\t&:focus {\n\t\t\tbackground: ", ";\n\t\t\tbox-shadow: ", ";\n\t\t\tborder-color: ", ";\n\t\t}\n\t\t&:active {\n\t\t\tbackground: ", ";\n\t\t}\n\t}\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n\tbackground: none;\n\tborder-color: transparent;\n\tcolor: ", ";\n\n\t&:hover,\n\t&:active,\n\t&:focus,\n\t&[data-focused='true'] {\n\t\tbackground-color: transparent;\n\t\ttext-decoration: underline;\n\t}\n\n\t&:active {\n\t\tcolor: ", ";\n\t}\n\n\t&[data-destructive='true'] {\n\t\tcolor: ", ";\n\n\t\t&:active {\n\t\t\tcolor: ", ";\n\t\t}\n\t}\n"]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteral(["\n\tbackground: none;\n\tborder-color: transparent;\n\tcolor: ", ";\n\n\t&:active {\n\t\tcolor: ", ";\n\t}\n\n\t&[data-destructive='true'] {\n\t\tcolor: ", ";\n\n\t\t&:active {\n\t\t\tcolor: ", ";\n\t\t}\n\t}\n"]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteral(["\n\tbackground-color: ", ";\n\tborder-color: ", ";\n\tcolor: ", ";\n\n\t&:hover {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t}\n\n\t&:focus,\n\t&[data-focused='true'] {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&:active {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&[data-destructive='true'] {\n\t\tcolor: ", ";\n\n\t\t&:hover,\n\t\t&:focus {\n\t\t\tborder-color: ", ";\n\t\t}\n\n\t\t&:active {\n\t\t\tcolor: ", ";\n\t\t}\n\t}\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteral(["\n\tbackground-color: ", ";\n\tborder-color: ", ";\n\tcolor: ", ";\n\n\t&:hover {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t}\n\n\t&:focus,\n\t&[data-focused='true'] {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&:active {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&[data-destructive='true'] {\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\n\t\t&:hover,\n\t\t&:active,\n\t\t&:focus,\n\t\t&[data-focused='true'] {\n\t\t\tborder-color: ", ";\n\t\t\tcolor: ", ";\n\t\t}\n\n\t\t&:active {\n\t\t\tcolor: ", ";\n\t\t}\n\t}\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteral(["\n\tbackground-color: ", ";\n\tborder-color: ", ";\n\tcolor: ", ";\n\n\t&:active {\n\t\tcolor: ", ";\n\t}\n\n\t&:hover {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&:focus,\n\t&[data-focused='true'] {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t\tcolor: ", ";\n\t}\n\n\t&:active {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\t}\n\n\t&[data-destructive='true'] {\n\t\tbackground-color: ", ";\n\t\tborder-color: ", ";\n\n\t\t&:hover,\n\t\t&:focus,\n\t\t&[data-focused='true'] {\n\t\t\tbackground-color: ", ";\n\t\t\tborder-color: ", ";\n\t\t}\n\n\t\t&:active {\n\t\t\tbackground-color: ", ";\n\t\t}\n\t}\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteral([""]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n\tmin-height: calc( ", " / 2 );\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n\tmin-height: calc( ", " / 2 );\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n\tmin-height: calc( ", " / 2 );\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n\tmin-height: ", ";\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n\tmin-height: ", ";\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n\t", ";\n\tmin-height: ", ";\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral([""]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n\tmin-height: ", ";\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n\tmin-height: ", ";\n\n\t&[data-icon='true'] {\n\t\tmin-width: ", ";\n\t}\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n\tcolor: ", ";\n\tfont-weight: 600;\n\tpadding-left: ", ";\n\tpadding-right: ", ";\n\n\t&:active {\n\t\tcolor: ", ";\n\t}\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

/**
 * External dependencies
 */
import { css, ui } from '@wp-g2/styles';
/**
 * Internal dependencies
 */

import * as baseButtonStyles from '../base-button/styles';
export var Button = css(_templateObject(), ui.get('buttonTextColor'), ui.get('buttonPaddingX'), ui.get('buttonPaddingX'), ui.get('buttonTextColorActive'));
export var xLarge = css(_templateObject2(), ui.get('controlHeightXLarge'), ui.get('controlHeightXLarge'));
export var large = css(_templateObject3(), ui.get('controlHeightLarge'), ui.get('controlHeightLarge'));
export var medium = css(_templateObject4());
export var small = css(_templateObject5(), ui.padding.x(ui.get('controlPaddingX')), ui.get('controlHeightSmall'), ui.get('controlHeightSmall'));
export var xSmall = css(_templateObject6(), ui.padding.x(ui.get('controlPaddingXSmall')), ui.get('controlHeightXSmall'), ui.get('controlHeightXSmall'));
export var xxSmall = css(_templateObject7(), ui.padding.x(ui.get('controlHeightXXSmall')), ui.get('controlHeightXXSmall'), ui.get('controlHeightXXSmall'));
export var half = css(_templateObject8(), ui.get('controlHeight'), ui.get('controlHeight'));
export var halfLarge = css(_templateObject9(), ui.get('controlHeightLarge'), ui.get('controlHeightLarge'));
export var halfSmall = css(_templateObject10(), ui.get('controlHeightSmall'), ui.get('controlHeightSmall'));
export var icon = css(_templateObject11(), ui.padding(0));
export var plain = css(_templateObject12());
export var primary = css(_templateObject13(), ui.get('buttonPrimaryColor'), ui.get('buttonPrimaryBorderColor'), ui.get('buttonPrimaryTextColor'), ui.get('buttonPrimaryTextColorActive'), ui.get('buttonPrimaryColorHover'), ui.get('buttonPrimaryBorderColorHover'), ui.get('buttonPrimaryTextColorHover'), ui.get('buttonPrimaryColorFocus'), ui.get('buttonPrimaryBorderColorFocus'), ui.get('buttonPrimaryTextColorFocus'), ui.get('buttonPrimaryColorActive'), ui.get('buttonPrimaryBorderColorActive'), ui.color.destructive, ui.color.destructive, ui.color.destructive, ui.color.destructive, ui.color.text);
export var secondary = css(_templateObject14(), ui.get('buttonSecondaryColor'), ui.get('buttonSecondaryBorderColor'), ui.get('buttonSecondaryTextColor'), ui.get('buttonSecondaryColorHover'), ui.get('buttonSecondaryBorderColorHover'), ui.get('buttonSecondaryColorFocus'), ui.get('buttonSecondaryBorderColorFocus'), ui.get('buttonSecondaryTextColorFocus'), ui.get('buttonSecondaryColorActive'), ui.get('buttonSecondaryBorderColorActive'), ui.get('buttonSecondaryTextColorActive'), ui.color.destructive, ui.color.destructive, ui.color.destructive, ui.color.destructive, ui.color.text);
export var tertiary = css(_templateObject15(), ui.get('buttonTertiaryColor'), ui.get('buttonTertiaryBorderColor'), ui.get('buttonTertiaryTextColor'), ui.get('buttonTertiaryColorHover'), ui.get('buttonTertiaryBorderColorHover'), ui.get('buttonTertiaryColorFocus'), ui.get('buttonTertiaryBorderColorFocus'), ui.get('buttonTertiaryTextColorFocus'), ui.get('buttonTertiaryColorActive'), ui.get('buttonTertiaryBorderColorActive'), ui.get('buttonTertiaryTextColorActive'), ui.color.destructive, ui.color.destructive, ui.color.text);
export var link = css(_templateObject16(), ui.get('linkColor'), ui.get('linkColorActive'), ui.color.destructive, ui.color.text);
export var plainLink = css(_templateObject17(), ui.padding.x(0), ui.get('linkColor'), ui.get('linkColorActive'), ui.color.destructive, ui.color.text);
export var subtle = baseButtonStyles.subtle;
export var control = baseButtonStyles.control;
var subtleControlActiveTransition = css({
  transition: ['all', ui.get('transitionDuration'), ui.get('transitionTimingFunctionControl')].join(' ')
});
export var subtleControlActive = css(_templateObject18(), subtleControlActiveTransition, ui.zIndex('ControlFocus', 1), ui.get('buttonControlActiveStateColor'), ui.get('buttonControlActiveStateColor'), ui.get('buttonControlActiveStateTextColor'), ui.get('buttonControlActiveStateColorHover'), ui.get('buttonControlActiveStateColorFocus'), ui.get('buttonControlActiveStateBoxShadowFocus'), ui.get('buttonControlActiveStateBorderColorFocus'), ui.get('buttonControlActiveStateColorActive'));
export var subtleControl = css(_templateObject19(), baseButtonStyles.subtleControl, subtleControlActive);
export var currentColor = baseButtonStyles.currentColor;
