// @wordpress/components/build/ui/surface/styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getBorders = getBorders;
exports.grid = exports.dotted = exports.tertiary = exports.secondary = exports.primary = exports.background = exports.Surface = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styles = require("@wp-g2/styles");

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground: ", ";\n\tbackground-image: ", ";\n\tbackground-size: ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground: ", ";\n\tbackground-size: ", ";\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2.default)([""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground-color: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tbackground-color: ", ";\n\tcolor: ", ";\n\tposition: relative;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Surface = (0, _styles.css)(_templateObject(), _styles.ui.get('surfaceColor'), _styles.ui.color.text);
exports.Surface = Surface;
var background = (0, _styles.css)(_templateObject2(), _styles.ui.get('surfaceBackgroundColor'));
/**
 * @param {Object} props
 * @param {boolean} [props.border]
 * @param {boolean} [props.borderBottom]
 * @param {boolean} [props.borderLeft]
 * @param {boolean} [props.borderRight]
 * @param {boolean} [props.borderTop]
 */

exports.background = background;

function getBorders(_ref) {
  var border = _ref.border,
      borderBottom = _ref.borderBottom,
      borderLeft = _ref.borderLeft,
      borderRight = _ref.borderRight,
      borderTop = _ref.borderTop;
  var borderStyle = "1px solid ".concat(_styles.ui.get('surfaceBorderColor'));
  return (0, _styles.css)({
    border: border ? borderStyle : undefined,
    borderBottom: borderBottom ? borderStyle : undefined,
    borderLeft: borderLeft ? borderStyle : undefined,
    borderRight: borderRight ? borderStyle : undefined,
    borderTop: borderTop ? borderStyle : undefined
  });
}

var primary = (0, _styles.css)(_templateObject3());
exports.primary = primary;
var secondary = (0, _styles.css)(_templateObject4(), _styles.ui.get('surfaceBackgroundTintColor'));
exports.secondary = secondary;
var tertiary = (0, _styles.css)(_templateObject5(), _styles.ui.get('surfaceBackgroundTertiaryColor'));
exports.tertiary = tertiary;
var customBackgroundSize = [_styles.ui.get('surfaceBackgroundSize'), _styles.ui.get('surfaceBackgroundSize')].join(' ');
var dottedBackground1 = ['90deg', [_styles.ui.get('surfaceBackgroundColor'), _styles.ui.get('surfaceBackgroundSizeDotted')].join(' '), 'transparent 1%'].join(',');
var dottedBackground2 = [[_styles.ui.get('surfaceBackgroundColor'), _styles.ui.get('surfaceBackgroundSizeDotted')].join(' '), 'transparent 1%'].join(',');
var dottedBackgroundCombined = ["linear-gradient( ".concat(dottedBackground1, " ) center"), "linear-gradient( ".concat(dottedBackground2, " ) center"), _styles.ui.get('surfaceBorderBoldColor')].join(',');
var dotted = (0, _styles.css)(_templateObject6(), dottedBackgroundCombined, customBackgroundSize);
exports.dotted = dotted;
var gridBackground1 = ["".concat(_styles.ui.get('surfaceBorderSubtleColor'), " 1px"), 'transparent 1px'].join(',');
var gridBackground2 = ['90deg', "".concat(_styles.ui.get('surfaceBorderSubtleColor'), " 1px"), 'transparent 1px'].join(',');
var gridBackgroundCombined = ["linear-gradient( ".concat(gridBackground1, " )"), "linear-gradient( ".concat(gridBackground2, " )")].join(',');
var grid = (0, _styles.css)(_templateObject7(), _styles.ui.get('surfaceBackgroundColor'), gridBackgroundCombined, customBackgroundSize);
exports.grid = grid;
