// @wordpress/block-editor/build-module/components/inserter/menu.native.js
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { useEffect, useCallback } from '@wordpress/element';
import { useSelect, useDispatch } from '@wordpress/data';
import { createBlock, rawHandler, store as blocksStore } from '@wordpress/blocks';
import { getClipboard } from '@wordpress/components';
/**
 * Internal dependencies
 */

import InserterSearchResults from './search-results';
import { store as blockEditorStore } from '../../store';

function InserterMenu(_ref) {
  var _onSelect = _ref.onSelect,
      onDismiss = _ref.onDismiss,
      rootClientId = _ref.rootClientId,
      clientId = _ref.clientId,
      isAppender = _ref.isAppender,
      shouldReplaceBlock = _ref.shouldReplaceBlock,
      insertionIndex = _ref.insertionIndex;

  var _useDispatch = useDispatch(blockEditorStore),
      showInsertionPoint = _useDispatch.showInsertionPoint,
      hideInsertionPoint = _useDispatch.hideInsertionPoint,
      clearSelectedBlock = _useDispatch.clearSelectedBlock,
      insertBlock = _useDispatch.insertBlock,
      removeBlock = _useDispatch.removeBlock,
      resetBlocks = _useDispatch.resetBlocks,
      insertDefaultBlock = _useDispatch.insertDefaultBlock;

  var _useSelect = useSelect(function (select) {
    var _select = select(blockEditorStore),
        getInserterItems = _select.getInserterItems,
        getBlockRootClientId = _select.getBlockRootClientId,
        getBlockSelectionEnd = _select.getBlockSelectionEnd,
        selectBlockEditorStore = _objectWithoutProperties(_select, ["getInserterItems", "getBlockRootClientId", "getBlockSelectionEnd"]);

    var targetRootClientId = rootClientId;

    if (!targetRootClientId && !clientId && !isAppender) {
      var end = getBlockSelectionEnd();

      if (end) {
        targetRootClientId = getBlockRootClientId(end) || undefined;
      }
    }

    return {
      items: getInserterItems(targetRootClientId),
      destinationRootClientId: targetRootClientId,
      getBlockOrder: selectBlockEditorStore.getBlockOrder,
      getBlockCount: selectBlockEditorStore.getBlockCount,
      canInsertBlockType: selectBlockEditorStore.canInsertBlockType
    };
  }),
      items = _useSelect.items,
      destinationRootClientId = _useSelect.destinationRootClientId,
      getBlockOrder = _useSelect.getBlockOrder,
      getBlockCount = _useSelect.getBlockCount,
      canInsertBlockType = _useSelect.canInsertBlockType;

  var _useSelect2 = useSelect(function (select) {
    return select(blocksStore);
  }),
      getBlockType = _useSelect2.getBlockType;

  useEffect(function () {
    if (shouldReplaceBlock) {
      var count = getBlockCount(); // Check if there is a rootClientId because that means it is a nested replaceable block
      // and we don't want to clear/reset all blocks.

      if (count === 1 && !rootClientId) {
        // Removing the last block is not possilble with `removeBlock` action.
        // It always inserts a default block if the last of the blocks have been removed.
        clearSelectedBlock();
        resetBlocks([]);
      } else {
        var blockToReplace = getBlockOrder(destinationRootClientId)[insertionIndex];
        removeBlock(blockToReplace, false);
      }
    } // Show/Hide insertion point on Mount/Dismount


    showInsertionPoint(destinationRootClientId, insertionIndex);
    return hideInsertionPoint;
  }, []);
  var onClose = useCallback(function () {
    // If should replace but didn't insert any block
    // re-insert default block.
    if (shouldReplaceBlock) {
      insertDefaultBlock({}, destinationRootClientId, insertionIndex);
    }

    onDismiss();
  }, [shouldReplaceBlock, destinationRootClientId, insertionIndex]);
  var onInsert = useCallback(function (item) {
    var name = item.name,
        initialAttributes = item.initialAttributes,
        innerBlocks = item.innerBlocks;
    var newBlock = createBlock(name, initialAttributes, innerBlocks);
    insertBlock(newBlock, insertionIndex, destinationRootClientId);
  }, [insertBlock, destinationRootClientId, insertionIndex]);
  /**
   * Processes the inserter items to check
   * if there's any copied block in the clipboard
   * to add it as an extra item
   */

  function getItems() {
    var _clipboardBlock;

    // Filter out reusable blocks (they will be added in another tab)
    var itemsToDisplay = items.filter(function (_ref2) {
      var name = _ref2.name;
      return name !== 'core/block';
    });
    var clipboard = getClipboard();
    var clipboardBlock = rawHandler({
      HTML: clipboard
    })[0];
    var canAddClipboardBlock = canInsertBlockType((_clipboardBlock = clipboardBlock) === null || _clipboardBlock === void 0 ? void 0 : _clipboardBlock.name, destinationRootClientId);

    if (!canAddClipboardBlock) {
      return itemsToDisplay;
    }

    var _getBlockType = getBlockType(clipboardBlock.name),
        icon = _getBlockType.icon,
        name = _getBlockType.name;

    var _clipboardBlock2 = clipboardBlock,
        initialAttributes = _clipboardBlock2.attributes,
        innerBlocks = _clipboardBlock2.innerBlocks;
    clipboardBlock = {
      id: 'clipboard',
      name: name,
      icon: icon,
      initialAttributes: initialAttributes,
      innerBlocks: innerBlocks
    };
    return [clipboardBlock].concat(_toConsumableArray(itemsToDisplay));
  }

  return createElement(InserterSearchResults, {
    onClose: onClose,
    items: getItems(),
    onSelect: function onSelect(item) {
      onInsert(item);

      _onSelect(item);
    }
  });
}

export default InserterMenu;
