// @wordpress/components/build-module/ui/surface/styles.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n\tbackground: ", ";\n\tbackground-image: ", ";\n\tbackground-size: ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n\tbackground: ", ";\n\tbackground-size: ", ";\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n\tbackground: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n\tbackground: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral([""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n\tbackground-color: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n\tbackground-color: ", ";\n\tcolor: ", ";\n\tposition: relative;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

/**
 * External dependencies
 */
import { css, ui } from '@wp-g2/styles';
export var Surface = css(_templateObject(), ui.get('surfaceColor'), ui.color.text);
export var background = css(_templateObject2(), ui.get('surfaceBackgroundColor'));
/**
 * @param {Object} props
 * @param {boolean} [props.border]
 * @param {boolean} [props.borderBottom]
 * @param {boolean} [props.borderLeft]
 * @param {boolean} [props.borderRight]
 * @param {boolean} [props.borderTop]
 */

export function getBorders(_ref) {
  var border = _ref.border,
      borderBottom = _ref.borderBottom,
      borderLeft = _ref.borderLeft,
      borderRight = _ref.borderRight,
      borderTop = _ref.borderTop;
  var borderStyle = "1px solid ".concat(ui.get('surfaceBorderColor'));
  return css({
    border: border ? borderStyle : undefined,
    borderBottom: borderBottom ? borderStyle : undefined,
    borderLeft: borderLeft ? borderStyle : undefined,
    borderRight: borderRight ? borderStyle : undefined,
    borderTop: borderTop ? borderStyle : undefined
  });
}
export var primary = css(_templateObject3());
export var secondary = css(_templateObject4(), ui.get('surfaceBackgroundTintColor'));
export var tertiary = css(_templateObject5(), ui.get('surfaceBackgroundTertiaryColor'));
var customBackgroundSize = [ui.get('surfaceBackgroundSize'), ui.get('surfaceBackgroundSize')].join(' ');
var dottedBackground1 = ['90deg', [ui.get('surfaceBackgroundColor'), ui.get('surfaceBackgroundSizeDotted')].join(' '), 'transparent 1%'].join(',');
var dottedBackground2 = [[ui.get('surfaceBackgroundColor'), ui.get('surfaceBackgroundSizeDotted')].join(' '), 'transparent 1%'].join(',');
var dottedBackgroundCombined = ["linear-gradient( ".concat(dottedBackground1, " ) center"), "linear-gradient( ".concat(dottedBackground2, " ) center"), ui.get('surfaceBorderBoldColor')].join(',');
export var dotted = css(_templateObject6(), dottedBackgroundCombined, customBackgroundSize);
var gridBackground1 = ["".concat(ui.get('surfaceBorderSubtleColor'), " 1px"), 'transparent 1px'].join(',');
var gridBackground2 = ['90deg', "".concat(ui.get('surfaceBorderSubtleColor'), " 1px"), 'transparent 1px'].join(',');
var gridBackgroundCombined = ["linear-gradient( ".concat(gridBackground1, " )"), "linear-gradient( ".concat(gridBackground2, " )")].join(',');
export var grid = css(_templateObject7(), ui.get('surfaceBackgroundColor'), gridBackgroundCombined, customBackgroundSize);
