// @wordpress/components/build/ui/control-label/styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.block = exports.inline = exports.small = exports.medium = exports.large = exports.ControlLabel = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styles = require("@wp-g2/styles");

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tdisplay: block;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tdisplay: inline-block;\n\tvertical-align: middle;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\tdisplay: inline-block;\n\tline-height: ", ";\n\tmargin: 0;\n\tmax-width: 100%;\n\tpadding-bottom: ", ";\n\tpadding-top: ", ";\n\n\t&:active {\n\t\tuser-select: none;\n\t}\n\n\t", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2.default)(["\n\t> * {\n\t\tposition: relative;\n\t\ttop: 0.5px;\n\t}\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var lineHeight = "calc(".concat(_styles.ui.get('fontSize'), " * 1.2)");
/* eslint-disable jsdoc/valid-types */

/**
 * @param {Parameters<typeof ui.get>[0]} size The padding size.
 */

/* eslint-enable jsdoc/valid-types */

function getPadding(size) {
  return "calc((".concat(_styles.ui.get(size), " - ").concat(lineHeight, ") / 2)");
}

var highDpiAdjust = (0, _styles.getHighDpi)(_templateObject());
var ControlLabel = (0, _styles.css)(_templateObject2(), lineHeight, getPadding('controlHeight'), getPadding('controlHeight'), highDpiAdjust);
exports.ControlLabel = ControlLabel;
var large = (0, _styles.css)(_templateObject3(), getPadding('controlHeightLarge'), getPadding('controlHeightLarge'));
exports.large = large;
var medium = (0, _styles.css)(_templateObject4(), getPadding('controlHeight'), getPadding('controlHeight'));
exports.medium = medium;
var small = (0, _styles.css)(_templateObject5(), getPadding('controlHeightSmall'), getPadding('controlHeightSmall'));
exports.small = small;
var inline = (0, _styles.css)(_templateObject6());
exports.inline = inline;
var block = (0, _styles.css)(_templateObject7());
exports.block = block;
